src/components/textarea/textarea.spec.js

```diff
@@ -65,6 +66,30 @@ describe("Textarea", () => {
     });
   });
 
+  it.each([
+    {
+      inputIcon: "search",
+    },
+    {
+      error: "error",
+    },
+    {
+      warning: "warning",
+    },
+    {
+      info: "info",
+    },
+  ])(
+    "pass hasIcon to InputPresentation when an icon is present inside",
+    (props) => {
+      wrapper = renderTextarea({
+        children: "mock content",
+        ...props,
+      });
+      expect(wrapper.find(StyledTextarea).props().hasIcon).toBe(true);
```

Comment:
It might be better to assertStyleMatch here instead of the prop value so it's easier when we come to migrate the file to RTL